time.go

```diff
@@ -65,7 +68,7 @@ func (t Time) MarshalJSON() ([]byte, error) {
 // UnmarshalJSON implements json.Unmarshaler.
 // It supports string and null input.
 func (t *Time) UnmarshalJSON(data []byte) error {
-	if len(data) > 0 && data[0] == 'n' {
+	if len(data) > 0 && data[0] == 'n' || bytes.Equal(data, []byte("NaT")) {{
```

Comment:
```suggestion
	if len(data) > 0 && data[0] == 'n' || bytes.Equal(data, []byte(`"NaT"`)) {
```

Reply:
Done thanks good catch ✅ and for tests should be good as well